app/models/general_info.rb

```diff
@@ -54,7 +54,7 @@ def incident_unique?
                                       incident_time_str: incident_time_str,
                                       incident_date_str: incident_date_str)
 
-    if general_infos.any? { |gi| gi.id != id && !gi.incident.try(:target).try(:deleted?) }
+    if general_infos.any? { |gi| gi.id != id && !gi.partial && !gi.incident.try(:target).try(:deleted?) }
```

Comment:
There's a lot of stuff on this line now, making it a little hard to understand what's going on.

Could you instead add `partial: false` to the `where` clause above?